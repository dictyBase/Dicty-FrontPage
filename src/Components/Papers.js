// @flow
import React from "react"
import { Flex } from "grid-styled"
import FontAwsome from "react-fontawesome"

import {
  PaperContainer,
  PaperTitle,
  Header,
  ListBox,
  ListItems,
  LeadText,
  MainContent,
  SourceContent,
  SourceTitle,
  Link,
  MoreLink,
} from "./styles"

/**
 * The Papers component that will be displayed on the right in every
 * react web application of [dictyBase](http://dictybase.org).
 */

const Papers = (props: {
  /** List of paper items */
  papers: Array<{
    "id": number,
    "author": string,
    "title": string,
    "journal": string,
  }>,
}) => {
  const text = props.papers.map((paper, index) =>
    <ListItems key={index}>
      <LeadText>
        {paper.author}
      </LeadText>
      <MainContent>
        <strong><em>{paper.title}</em></strong>
      </MainContent>
      <SourceContent>
        <SourceTitle>Journal: </SourceTitle>
          {paper.journal}
        <Link href={paper.link} target="new"> Pubmed</Link>
      </SourceContent>
    </ListItems>,
  )

  return (
    <PaperContainer>
      <Header>
        <Flex wrap>
            <PaperTitle>
                <FontAwsome name="paperclip fa-lg" />
            </PaperTitle>
            <PaperTitle>
                LATEST PAPERS
            </PaperTitle>
        </Flex>
      </Header>
      <ListBox>
        {text}
      </ListBox>
      <MoreLink>
        <FontAwsome name="plus" />
        <Link href="" alt="more papers">
          {" "}more papers{" "}
        </Link>
      </MoreLink>
    </PaperContainer>
  )
}

export default Papers
