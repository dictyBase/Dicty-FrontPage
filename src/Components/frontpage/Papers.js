// @flow
import React from "react"
import { Flex } from "rebass"
import FontAwesome from "react-fontawesome"

import {
  PaperContainer,
  PaperTitle,
  Header,
  ListBox,
  ListItems,
  LeadText,
  MainContent,
  SourceContent,
  SourceTitle,
  Link,
  MoreLink,
} from "../styles"

/**
 * The Papers component that will be displayed on the right in every
 * react web application of [dictyBase](http://dictybase.org).
 */

const Papers = (props: {
  /** List of paper items */
  papers: Array<{
    id: number,
    author: string,
    title: string,
    journal: string,
    link: string,
  }>,
}) => {
  const text = props.papers.map((paper, index) => (
    <ListItems key={index}>
      <LeadText>{paper.author}</LeadText>
      <MainContent>
        <strong>
          <em>{paper.title}</em>
        </strong>
      </MainContent>
      <SourceContent>
        <SourceTitle>Journal: </SourceTitle>
        {paper.journal}
        <Link href={paper.link} target="new">
          {" "}
          Pubmed
        </Link>
      </SourceContent>
    </ListItems>
  ))

  return (
    <PaperContainer>
      <Header>
        <Flex wrap>
          <PaperTitle>
            <FontAwesome name="paperclip fa-lg" />
          </PaperTitle>
          <PaperTitle>LATEST PAPERS</PaperTitle>
        </Flex>
      </Header>
      <ListBox>{text}</ListBox>
      <MoreLink>
        <FontAwesome name="plus" />
        <Link href="" alt="more papers">
          {" "}
          more papers{" "}
        </Link>
      </MoreLink>
    </PaperContainer>
  )
}

export default Papers
