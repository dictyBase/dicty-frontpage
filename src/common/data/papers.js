module.exports = [
  {
    author: "Tanaka, Jahan, Kondo, Nakano & Yumura (2019)",
    title:
      "Cytokinesis D is Mediated by Cortical Flow of Dividing Cells Instead of Chemotaxis.",
    journal: "Cells 8",
    link: "https://www.ncbi.nlm.nih.gov/pubmed/31108912",
  },
  {
    author:
      "Odell, Sikirzhytski, Tikhonenko, Cobani, Khodjakov & Koonce (2019)",
    title:
      "Force balances between interphase centrosomes, as revealed by laser ablation.",
    journal: "Mol. Biol. Cell mbcE19010034",
    link: "https://www.ncbi.nlm.nih.gov/pubmed/31067156",
  },
  {
    author:
      "Chen, Luck, Rabe, Dinh, Shaulsky, Nelson, Gershenzon, Dickschat, Kollner & Chen (2019)",
    title:
      "A terpene synthase-cytochrome P450 cluster in Dictyostelium discoideum produces a novel trisnorsesquiterpene.",
    journal: "Elife 8",
    link: "https://www.ncbi.nlm.nih.gov/pubmed/31063135",
  },
  {
    author: "Yamada & Schaap (2019)",
    title:
      "Cyclic AMP induction of Dictyostelium prespore gene expression requires autophagy.",
    journal: "Dev. Biol.",
    link: "https://www.ncbi.nlm.nih.gov/pubmed/31051160",
  },
  {
    author: "Saga, Iwade, Araki, Ishikawa & Kawata (2019)",
    title: "Analysis of DrkA kinase's role in STATa activation.",
    journal: "Genes Cells",
    link: "https://www.ncbi.nlm.nih.gov/pubmed/31002205",
  },
]
