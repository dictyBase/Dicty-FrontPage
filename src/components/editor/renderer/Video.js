import React, { PureComponent } from "react"

export default class Video extends PureComponent {
  onClick = e => {
    e.stopPropagation()
  }

  render() {
    const { node, isSelected, editor, attributes } = this.props
    const src = node.data.get("src")
    const url = node.data.get("url")

    const { readOnly } = editor.props

    const wrapperStyle = {
      position: "relative",
      paddingBottom: "50.66%",
      paddingTop: "25px",
      height: "0",
      outline: isSelected ? `2px solid #0017ff` : "none",
    }

    const maskStyle = {
      display: isSelected ? "none" : "block",
      position: "absolute",
      top: "0px",
      left: "0px",
      right: "0px",
      bottom: "0px",
      height: "100%",
      cursor: "cell",
      zIndex: 1,
    }

    const iframeStyle = {
      position: "absolute",
      top: "0px",
      left: "0px",
      width: "100%",
      height: "100%",
    }

    return (
      <div>
        <div {...attributes} style={wrapperStyle}>
          <div style={readOnly ? {} : maskStyle} />
          <iframe
            title="video-embed"
            id="ytplayer"
            type="text/html"
            width="580"
            height="390"
            src={src}
            frameBorder="0"
            style={iframeStyle}
          />
        </div>
      </div>
    )
  }
}
