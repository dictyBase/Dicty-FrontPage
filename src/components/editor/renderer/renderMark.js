// @flow
import React, { type Node } from "react"
import { type Mark } from "slate"
import MARKS from "components/editor/constants/marks"

type Props = {
  children: Node,
  mark: Mark,
}

const renderMark = (props: Props) => {
  const { children, mark } = props
  switch (mark.type) {
    case MARKS.BOLD:
      return <strong>{children}</strong>

    case MARKS.ITALIC:
      return <em>{children}</em>

    case MARKS.UNDERLINE:
      return <u>{children}</u>

    case MARKS.STRIKETHROUGH:
      return <del>{children}</del>

    default:
      return null
  }
}

export default renderMark
