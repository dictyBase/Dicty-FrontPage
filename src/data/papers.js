module.exports = [
  {
    author: "Xie X, Gutierrez A, Trofimov V, Szilágyi I, Soldati T, Bakker E.",
    title:
    "<strong><em>Charged Solvatochromic Dyes as Signal Transducers in pH Independent Fluorescent and Colorimetric Ion Selective Nanosensors.</em></strong>",
    journal: "Anal Chem. 2015 Sep 9.[Epub ahead of print] <a href='https://www.ncbi.nlm.nih.gov/pubmed/26352133' target='new'>Pubmed</a>",
  },
  {
    author: "Schmith A, Spaller T, Gaube F, Fransson Å, Boesler B, Ojha S, Nellen W, Hammann C, Söderbom F, Winckler T.",
    title:
    "<strong><em>A host factor supports retrotransposition of the TRE5-A population in Dictyostelium cells by suppressing an Argonaute protein.</em></strong>",
    journal: "Mob DNA. 2015;6:14. doi: 10.1186/s13100-015-0045-5. <a href='https://www.ncbi.nlm.nih.gov/pubmed/26339297' target='new'>Pubmed</a> ",
  },
  {
    author: "Von Bülow J, Beitz E.",
    title:
    "<strong><em>Number and Regulation of Protozoan Aquaporins Reflect Environmental Complexity.</em></strong>",
    journal: "Biol Bull. 2015 Aug;229(1):38-46. <a href='https://www.ncbi.nlm.nih.gov/pubmed/26338868' target='new'>Pubmed</a> ",
  },
  {
    author: "Santarriaga S, Petersen A, Ndukwe K, Brandt A, Gerges N, Scaglione JB, Scaglione KM.",
    title:
    "<strong><em>The Social Amoeba Dictyostelium discoideum is Highly Resistant to Polyglutamine Aggregation.</em></strong>",
    journal: "J Biol Chem. 2015 Sep 1. doi:pii: jbc.M115.676247. [Epub ahead of print]<a href='https://www.ncbi.nlm.nih.gov/pubmed/26338868' target='new'>Pubmed</a> ",
  },
  {
    author: "Kim MK, Kim JH, Kim JS, Kang SO.",
    title:
    "<strong><em>Structure of the 34 kDa F-actin-bundling protein ABP34 from Dictyostelium discoideum.</em></strong>",
    journal: "Acta Crystallogr D Biol Crystallogr. 2015 Sep 1;71(Pt 9):1835-49. doi: 10.1107/S139900471501264X. Epub 2015 Aug 25. <a href='https://www.ncbi.nlm.nih.gov/pubmed/26338868' target='new'>Pubmed</a> ",
  },
  
]
