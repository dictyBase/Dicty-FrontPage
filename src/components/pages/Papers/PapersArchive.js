// @flow
import React from "react"
import Grid from "@material-ui/core/Grid"
import { withStyles } from "@material-ui/core"
import FontAwesome from "react-fontawesome"
import { Link } from "styles"
import papers from "data/papers"

const styles = theme => ({
  item: {
    paddingBottom: "10px",
  },
  banner: {
    minHeight: "150px",
    textAlign: "center",
    padding: "48px 30px 48px 30px",
    backgroundColor: "#E0F2F7",
  },
  header: {
    backgroundColor: "#E0F2F7",
    "@media (minWidth: 768px)": {
      fontSize: "63px",
      padding: "2px",
      margin: "2px",
    },
  },
  headerText: {
    fontSize: "21px",
  },
})

type Props = {
  /** Material-UI styling */
  classes: Object,
}

/**
 * This displays the Dicty papers archive.
 */

const PapersArchive = (props: Props) => {
  const { classes } = props

  const paperList = papers.map(paper => (
    <Grid container justify="center" key={paper.title}>
      <Grid item className={classes.item} xs={11}>
        <div>{paper.author}</div>
        <div>
          <strong>{paper.title}</strong>
        </div>
        <div>
          <em>{paper.journal}</em>
          <Link href={paper.link} target="new">
            {" "}
            Pubmed
          </Link>
        </div>
      </Grid>
    </Grid>
  ))

  return (
    <>
      <div className={classes.banner}>
        <h1 className={classes.header}>
          <FontAwesome name="paperclip" /> Dicty Papers
        </h1>
        <p className={classes.headerText}>
          Papers on <em>Dictyostelium</em> in the last 5 years
        </p>
      </div>
      <br />
      <Grid container justify="center">
        <Grid item>{paperList}</Grid>
      </Grid>
    </>
  )
}

export default withStyles(styles)(PapersArchive)
