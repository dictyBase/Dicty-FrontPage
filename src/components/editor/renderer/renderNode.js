// @flow
import React, { type Node } from "react"
import { type Block } from "slate"
import styled from "styled-components"
import Video from "components/editor/renderer/Video"
import BLOCKS from "components/editor/constants/blocks"
import INLINES from "components/editor/constants/inlines"

type Props = {
  node: Block,
  children: Node,
  attributes: Object,
}

const Image = styled.img`
  display: block;
  max-width: 100%;
  max-height: 20em;
  box-shadow: ${props => (props.selected ? "0 0 0 2px blue;" : "none")};
`

const renderNode = (props: Props) => {
  const { attributes, children, node } = props
  switch (node.type) {
    case BLOCKS.UL_LIST:
      return <ul {...attributes}>{children}</ul>

    case BLOCKS.OL_LIST:
      return <ol {...attributes}>{children}</ol>

    case BLOCKS.LIST_ITEM:
      return <li {...attributes}>{children}</li>

    case BLOCKS.HEADING_1:
      return <h1 {...attributes}>{children}</h1>

    case BLOCKS.HEADING_2:
      return <h2 {...attributes}>{children}</h2>

    case BLOCKS.HEADING_3:
      return <h3 {...attributes}>{children}</h3>

    case BLOCKS.BLOCKQUOTE:
      return <blockquote {...attributes}>{children}</blockquote>

    case INLINES.LINK: {
      const { data } = node
      const href = data.get("href")
      return (
        <a {...attributes} href={href}>
          {children}
        </a>
      )
    }

    case BLOCKS.IMAGE: {
      const { data, isSelected } = node
      const src = data.get("src")
      const className = isSelected ? "selected" : "unselected"
      const style = { display: "inline-block" }
      return (
        <img
          src={src}
          className={className}
          {...attributes}
          style={style}
          alt=""
        />
      )
    }

    case BLOCKS.ALIGN_LEFT:
      return (
        <div style={{ textAlign: "left" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_CENTER:
      return (
        <div style={{ textAlign: "center" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_RIGHT:
      return (
        <div style={{ textAlign: "right" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.HR:
      return <hr {...attributes} />

    case BLOCKS.TABLE:
      return (
        <table {...attributes}>
          <tbody>{children}</tbody>
        </table>
      )

    case BLOCKS.TABLE_ROW:
      return <tr {...attributes}>{children}</tr>

    case BLOCKS.TABLE_CELL:
      return <td {...attributes}>{children}</td>

    case BLOCKS.VIDEO:
      return <Video {...props} />

    default:
      return null
  }
}

export default renderNode
