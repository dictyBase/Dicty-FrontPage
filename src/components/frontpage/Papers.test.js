import React from "react"
import renderer from "react-test-renderer"
import Papers from "components/frontpage/Papers"
import papers from "data/papers"
import "jest-styled-components"

test("Papers snapshot test", () => {
  const component = renderer.create(<Papers papers={papers} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
