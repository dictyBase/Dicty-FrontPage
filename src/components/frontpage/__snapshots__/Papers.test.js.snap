// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Papers snapshot test 1`] = `
.c3 {
  box-sizing: border-box;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  line-height: 1.428;
  text-align: left;
  padding-left: 10px;
  padding-right: 10px;
  background-color: #eff8fb;
  border-radius: 15px;
  margin-bottom: 10px;
}

.c1 {
  color: black;
  font-size: 20px;
  padding: 15px 30px 0px 35px;
  vertical-align: top;
  text-align: right;
}

.c4 {
  padding-left: 5px;
  color: #086a87;
  font-size: 20px;
  vertical-align: top;
  text-align: left;
}

.c5 {
  padding: 0px 25px 10px 25px;
  font-size: 12px;
  margin-bottom: 5px;
  padding-bottom: 10px;
  margin-top: 12px;
}

.c6 {
  list-style-type: none;
  margin-bottom: 10px;
}

.c7 {
  color: #0b3861;
  padding-right: 10px;
}

.c8 {
  padding-right: 10px;
}

.c9 {
  color: #0b3861;
}

.c10 {
  padding-top: 7px;
  font-weight: bold;
  text-align: center;
}

.c11 {
  color: #428bca;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c12 {
  color: #0b3861;
  font-size: 11px;
  font-style: italic;
  font-weight: normal;
  text-align: center;
  padding-bottom: 15px;
}

@media (max-width:767px) {
  .c1 {
    font-size: 24px;
    text-align: right;
    padding: 20px 5px 20px 15px;
  }
}

@media (max-width:992px) and (min-width:767px) {
  .c5 {
    font-size: 10px;
  }
}

@media (max-width:768px) {
  .c5 {
    font-size: 16px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2 c3"
      is={null}
    >
      <span
        className="c4"
      >
        <span
          aria-hidden={true}
          className="fa fa-paperclip fa-lg"
        />
      </span>
      <span
        className="c4"
      >
        LATEST PAPERS
      </span>
    </div>
  </div>
  <ul
    className="c5"
  >
    <li
      className="c6"
    >
      <span
        className="c7"
      >
        Xie X, Gutierrez A, Trofimov V, Szilágyi I, Soldati T, Bakker E.
      </span>
      <span
        className="c8"
      >
        <strong>
          <em>
            Charged Solvatochromic Dyes as Signal Transducers in pH Independent Fluorescent and Colorimetric Ion Selective Nanosensors.
          </em>
        </strong>
      </span>
      <span
        className="c9"
      >
        <span
          className="c10"
        >
          Journal: 
        </span>
        Anal Chem. 2015 Sep 9.[Epub ahead of print]
        <a
          className="c11"
          href="https://www.ncbi.nlm.nih.gov/pubmed/26352133"
          target="new"
        >
           
          Pubmed
        </a>
      </span>
    </li>
    <li
      className="c6"
    >
      <span
        className="c7"
      >
        Schmith A, Spaller T, Gaube F, Fransson Å, Boesler B, Ojha S, Nellen W, Hammann C, Söderbom F, Winckler T.
      </span>
      <span
        className="c8"
      >
        <strong>
          <em>
            A host factor supports retrotransposition of the TRE5-A population in Dictyostelium cells by suppressing an Argonaute protein.
          </em>
        </strong>
      </span>
      <span
        className="c9"
      >
        <span
          className="c10"
        >
          Journal: 
        </span>
        Mob DNA. 2015;6:14. doi: 10.1186/s13100-015-0045-5.
        <a
          className="c11"
          href="https://www.ncbi.nlm.nih.gov/pubmed/26339297"
          target="new"
        >
           
          Pubmed
        </a>
      </span>
    </li>
    <li
      className="c6"
    >
      <span
        className="c7"
      >
        Von Bülow J, Beitz E.
      </span>
      <span
        className="c8"
      >
        <strong>
          <em>
            Number and Regulation of Protozoan Aquaporins Reflect Environmental Complexity.
          </em>
        </strong>
      </span>
      <span
        className="c9"
      >
        <span
          className="c10"
        >
          Journal: 
        </span>
        Biol Bull. 2015 Aug;229(1):38-46.
        <a
          className="c11"
          href="https://www.ncbi.nlm.nih.gov/pubmed/26338868"
          target="new"
        >
           
          Pubmed
        </a>
      </span>
    </li>
    <li
      className="c6"
    >
      <span
        className="c7"
      >
        Santarriaga S, Petersen A, Ndukwe K, Brandt A, Gerges N, Scaglione JB, Scaglione KM.
      </span>
      <span
        className="c8"
      >
        <strong>
          <em>
            The Social Amoeba Dictyostelium discoideum is Highly Resistant to Polyglutamine Aggregation.
          </em>
        </strong>
      </span>
      <span
        className="c9"
      >
        <span
          className="c10"
        >
          Journal: 
        </span>
        J Biol Chem. 2015 Sep 1. doi:pii: jbc.M115.676247. [Epub ahead of print]
        <a
          className="c11"
          href="https://www.ncbi.nlm.nih.gov/pubmed/26338868"
          target="new"
        >
           
          Pubmed
        </a>
      </span>
    </li>
    <li
      className="c6"
    >
      <span
        className="c7"
      >
        Kim MK, Kim JH, Kim JS, Kang SO.
      </span>
      <span
        className="c8"
      >
        <strong>
          <em>
            Structure of the 34 kDa F-actin-bundling protein ABP34 from Dictyostelium discoideum.
          </em>
        </strong>
      </span>
      <span
        className="c9"
      >
        <span
          className="c10"
        >
          Journal: 
        </span>
        Acta Crystallogr D Biol Crystallogr. 2015 Sep 1;71(Pt 9):1835-49. doi: 10.1107/S139900471501264X. Epub 2015 Aug 25.
        <a
          className="c11"
          href="https://www.ncbi.nlm.nih.gov/pubmed/26338868"
          target="new"
        >
           
          Pubmed
        </a>
      </span>
    </li>
  </ul>
  <div
    className="c12"
  >
    <span
      aria-hidden={true}
      className="fa fa-plus"
    />
    <a
      alt="more papers"
      className="c11"
      href="/papers"
    >
       
      more papers
       
    </a>
  </div>
</div>
`;
